rotkehlchen/tests/unit/decoders/test_weth.py

```diff
@@ -0,0 +1,231 @@
+            notes='Withdraw 0.5 WETH for ETH',
+            counterparty=CPT_WETH,
+        ), HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier(
+                '0x1f3aa6f7d33bfaaaf9cdd92b16fecdf911341601c02ad89b4ec0b80c66c28a07',
+            ),
+            sequence_index=2,
+            timestamp=1666256147000,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(0.5)),
+            location_label='0x4b2975AfF4DeF34D3Cd4f4759b45faF738D790D3',
+            notes='Receive 0.5 ETH',
+            counterparty=CPT_WETH,
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.parametrize('ethereum_accounts', [['0xDea6866A866C60d68fFDFc6178C12fCFdb9d0D47']])  # noqa: E501
+def test_weth_interaction_with_protocols(database, ethereum_manager):
+    """
+    Data for deposit is taken from
+    https://etherscan.io/tx/0x4a811e8cfa58cb5bd57d92d62e1f01c8578859705243fe69c6bd9e59f3dcd167
+    """
+    msg_aggregator = MessagesAggregator()
+    tx_hex = '0x4a811e8cfa58cb5bd57d92d62e1f01c8578859705243fe69c6bd9e59f3dcd167'
+    evmhash = deserialize_evm_tx_hash(tx_hex)
+    events = get_decoded_events_of_transaction(
+        ethereum_manager=ethereum_manager,
+        database=database,
+        msg_aggregator=msg_aggregator,
+        tx_hash=evmhash,
+    )
+    assert len(events) == 4
+    expected_events = [
+        HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier(
+                '0x4a811e8cfa58cb5bd57d92d62e1f01c8578859705243fe69c6bd9e59f3dcd167',
+            ),
+            sequence_index=0,
+            timestamp=1666284551000,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.011940359686863452')),
+            location_label='0xDea6866A866C60d68fFDFc6178C12fCFdb9d0D47',
+            notes='Burned 0.011940359686863452 ETH in gas from 0xDea6866A866C60d68fFDFc6178C12fCFdb9d0D47',  # noqa: E501
+            counterparty=CPT_GAS,
+        ), HistoryBaseEntry(
+            event_identifier=HistoryBaseEntry.deserialize_event_identifier(
+                '0x4a811e8cfa58cb5bd57d92d62e1f01c8578859705243fe69c6bd9e59f3dcd167',
+            ),
+            sequence_index=1,
+            timestamp=1666284551000,
+            location=Location.BLOCKCHAIN,
+            event_type=HistoryEventType.WITHDRAWAL,
+            event_subtype=HistoryEventSubType.REMOVE_ASSET,
+            asset=A_WETH,
+            balance=Balance(amount=FVal('0.764522981784947382')),
+            location_label='0xC36442b4a4522E871399CD717aBDD847Ab11FE88',
+            notes='Withdraw 0.764522981784947382 WETH for ETH',
+            counterparty=CPT_WETH,
```

Comment:
This shouldn't appear since is not an event of the user. The user only received ETH